File: 백준/Silver/2578. 빙고/빙고.java
import java.util.*;
import java.io.*;

public class Main {
	
	static int[][] map = new int[5][5];
	static int cnt_total;
	static int num;
	
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		cnt_total = 0;
		num = 0;
		
		for(int i=0; i<5; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			for(int j=0; j<5; j++) {
				map[i][j] = Integer.parseInt(st.nextToken());
			}
		}
		
		for(int i=0; i<5; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			for(int j=0; j<5; j++) {
				num++;
				Bingo(Integer.parseInt(st.nextToken()));
				Wid();
				Hei();
				Dia_right();
				Dia_left();

			    if(cnt_total >= 3) {
			    	System.out.println(num);
			    	return;
		    	}
                cnt_total = 0;
			}
		}
	}
	
	public static void Bingo(int n) {
		for(int i=0; i<5; i++) {
			for(int j=0; j<5; j++) {
				if(map[i][j] == n) {
					map[i][j] = 0;
				}
			}
		}
	}
	
	public static void Wid() {
		for(int i=0; i<5; i++) {
			int count = 0;
			for(int j=0; j<5; j++) {
				if(map[i][j] == 0) 
					count++;	
			}
			if(count == 5) 
				cnt_total++;
		}
	}
	
	public static void Hei() {
		for(int i=0; i<5; i++) {
			int count = 0;
			for(int j=0; j<5; j++) {
				if(map[j][i] == 0) 
					count++;
			}
			if(count == 5) 
				cnt_total++;
		}
	}
	
	public static void Dia_right() {
		int count = 0;		
		for(int i=0; i<5; i++) {
			if(map[i][i] == 0) 
				count++;
		}
		if(count == 5) 
			cnt_total++;
	}
	
	public static void Dia_left() {
		int count = 0;		
		for(int i=0; i<5; i++) {
			if(map[i][4-i] == 0)
				count++;
		}
		if(count == 5) 
			cnt_total++;
	}
}

File: 백준/Silver/2578. 빙고/빙고.cc
#include <iostream>
#include <algorithm>
#include <string>
#include <vector>


using namespace std;
int bingo[5][5];
int map[5][5];
int cnt_total, num;

void Wid() {
	for (int i = 0; i < 5; i++) {
		int cnt = 0;
		for (int j = 0; j < 5; j++) {
			if (bingo[i][j] == 0)
				cnt++;
		}
		if (cnt == 5) {
			cnt_total++;
		}
	}
}

void Hei() {
	for (int i = 0; i < 5; i++) {
		int cnt = 0;
		for (int j = 0; j < 5; j++) {
			if (bingo[j][i] == 0)
				cnt++;
		}
		if (cnt == 5) {
			cnt_total++;
		}
	}
}

void Dia_right() {
	int cnt = 0;
	for (int i = 0; i < 5; i++) {
		if (bingo[i][i]==0) 
			cnt++;
	}
	if (cnt == 5) {
		cnt_total++;
	}
}

void Dia_left() {
	int cnt = 0;
	for (int i = 0; i < 5; i++) {
		if (bingo[4-i][i] == 0)
			cnt++;
	}

	if (cnt == 5) {
		cnt_total++;
	}
}

int Bingo(int n) {
	for (int i = 0; i < 5; i++) {
		for (int j = 0; j < 5; j++) {
			if (bingo[i][j] == n) {
				bingo[i][j] = 0;
			}
		}
	}
	return 0;
}

int main() {

	int n;
	num = 0;

	for (int i = 0; i < 5; i++) {
		for (int j = 0; j < 5; j++) {
			cin >> bingo[i][j];
		}
	}

	for (int i = 0; i < 5; i++) {
		for (int j = 0; j < 5; j++) {
			num++;
			cin >> n;
			Bingo(n);
			Wid();
			Hei();
			Dia_right();
			Dia_left();

			if (cnt_total >= 3) {
				cout << num;
				return 0;
			}
			cnt_total = 0;
		}
	}
}
